src/lib/metricsMgr/MetricsManager.cpp

```diff
@@ -131,15 +183,57 @@ void MetricsManager::add(const std::string& srv, const std::string& subServ, con
     return;
   }
 
+  if (serviceValid(srv) == false)
+  {
+    return;
+  }
+
+  char* toFree      = strdup(subServ.c_str());
```

Comment:
Discussed at skype: try to move the "SP calculation logic" as part of subServiceValid() funciton (changing the function a bit... it should return the pointer to the SP char-string to use, probably the name should be also adjusted).